_posts/2020-09-07-introduction-to-system-tests.md

```diff
@@ -0,0 +1,258 @@
+@Test
+void testReceiveSimpleMessage() {
+    KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+    kafkaBridgeClientJob.consumerStrimziBridge().done();
+
+    // Send messages to Kafka
+    InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+        .withTopicName(TOPIC_NAME)
+        .withNamespaceName(NAMESPACE)
+        .withClusterName(CLUSTER_NAME)
+        .withMessageCount(MESSAGE_COUNT)
+        .withKafkaUsername(USER_NAME)
+        .withUsingPodName(kafkaClientsPodName)
+        .build();
+
+    assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
+
+    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+}
+
+@BeforeAll
+void createClassResources() throws Exception {
+    deployClusterOperator(NAMESPACE);
+    LOGGER.info("Deploy Kafka and KafkaBridge before tests");
+    // Deploy kafka
+    KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
+
+    KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+
+    kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+    // Deploy http bridge
+    KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
+        .editSpec()
+            .withNewConsumer()
+                .addToConfig(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+            .endConsumer()
+        .endSpec()
+        .done();
+}
+```
+
+This test case as you can see ensures that we are able to receive messages from `KafkaBridge` component.
+In the setup phase we deploy `cluster-operator`, one node Kafka cluster and also the `KafkaBridge` itself.
+Everything is inside the real Kubernetes cluster.
+All test cases with related test suites can be found in the `system-test` module to make it more sense.
+What is also worth to mentioned that our system test currently takes around ~30 hours but we have 3 separate sub-sets, which
+runs in parallel so it is "only" approximately 10h per each sub-set.
+Additionally, there is also a mvn profile for the main groups - `acceptance`, `regression`, `smoke`, `bridge` and `all`,
+but we suggest to use profile with id `all` (default) and then include or exclude specific groups.
+If you want specify the profile, use the `-P` flag for example `-Psmoke`.
+All available test groups are listed in [Constants](https://github.com/strimzi/strimzi-kafka-operator/blob/master/systemtest/src/main/java/io/strimzi/systemtest/Constants.java)
+class.
+
+## How they run with relation with Kubernetes
+
+Basically, what I mentioned in previous section system tests run against real Kubernetes cluster.
+You can run these tests against `Minikube`, `Minishift`, big `Kubernetes` or `Openshift` cluster.
+In the `test` module we have our representation of cluster which decide on which environment it is running.
+Moreover, we are using `Azure pipelines` to trigger these tests and using `Minikube`.
```

Comment:
It would be better to describe for what we use AZP and how the tests are split there.